### ISSUE TITLE
New expense flow: allow negative amounts for items

### ISSUE THREAD

**Issue by Betree** [opencollective/opencollective#3044]
## User story 
As a professional submitting an expense on Open Collective, I can add items with negative amounts to reflect discounts or adjustments on the total amount.

For example this month I wanted to reflect the fact that I got a double transfer last month by subtracting it on my invoice. The app that I use for my accountability, https://shine.fr, let me do that but not Open Collective.

**Shine**

![image](https://user-images.githubusercontent.com/1556356/78888729-7fef6300-7a62-11ea-82ca-a32b2ff20f45.png)

**Open Collective**

![image](https://user-images.githubusercontent.com/1556356/78888573-2edf6f00-7a62-11ea-8ffe-1eec61924871.png)

## Best solution for this problem
Allow negative amounts for items, but add a validation to make sure that the total is still greater than 0.

**Comment by contraexemplo** [comment]
I really like this. How do you do that on Shine? Is it a matter of just adding a minus sign?

**Comment by Betree** [comment]
@contraexemplo it is, I made a quick screencast to showcase the flow: 
https://drive.google.com/file/d/12spd1Ef2Xu8pKUpkV_J62QaKPvjVfMZB/view?usp=drivesdk

**Comment by contraexemplo** [comment]
@Betree Oooh that's really nice! I like that flow.

**Comment by Betree** [comment]
@piamancini @alanna do you think this one is compatible with the different legislations around?

**Comment by alanna** [comment]
Yes it's fine to have negative line items on invoices, as long as the sum total doesn't go into negative.

**Comment by Betree** [comment]
It seems that we already allow that on the API when we generate the monthly host invoice, so chances are we only need to allow that in the frontend.
![image](https://user-images.githubusercontent.com/1556356/120000659-56c9c300-bfd3-11eb-8793-40f55d4ed24f.png)

**Comment by Betree** [comment]
Related: we may want to allow items with `0` amount, it's quite common to see that on invoices

![image](https://github.com/opencollective/opencollective/assets/1556356/ca76948e-1256-481c-812f-2fb225a6954c)
